### DISCUSSION TITLE
New Sonoff Basic GPIO Query

### DISCUSSION THREAD

**Discussion by muzzthegreat** [arendst/Tasmota#12289] [Q&A] [answered]
Hi there,
I am using the new Sonoff basic RF R2 (I think) for my fish tank. The switch turns my lights on and I wanted to connect a DS18B20 temperature probe and a level switch (to check when my water level is low). 
I know the GPIO pin issue, so I want to connect the temp probe on GPIO1 (TX) as I know it has boot failure if pulled low. I believe the sensor won't create this failure.
Then I want to connect the level switch to GPIO3 (RX) so it won't interfere with the startup.
I've set GPIO1 to the DS18B20 and GPIO3 to Switch 2 in the Configuration and I believe this should work, but when I plug the device in, the blue light keeps flashing and then when I toggle the level switch (on GPIO3) it turns the relay on and off, but the blue light still flashes and it isn't detected in tasmota.
If I disconnect the level switch, then the device does work correctly and the temperature probe is visible, so I don't believe the temperature probe is causing any issues.
Not sure why it is doing this? Does anyone have any advice for me?
Thanks

**Comment by Jason2866** [comment, marked as answer]
Use GPIO 2 or GPIO 14 as shown here. https://tasmota.github.io/docs/devices/Sonoff-Basic/ Do not use the troublemaker GPIO

**Reply by sfromis** [reply]
Try doing as little as possible at a time (skip the sensor to test), and be clear about exact details of what you're doing with gpio3, like voltage levels.

**Reply by muzzthegreat** [reply]
The level switch on GPIO3 is just a switch, with a gnd and a signal cable. I've checked the continuity and it goes on and off when toggled.
Attached are pics of my device. The one I am using, I have soldered pins on both of the sides (J1 - gnd, led, key, 3v3) & (J2 - gnd, TX, RX, 3v3). 

So what I can see from this device is that I can only get pins for GPIO1, 2 & 3 and GPIO1 & 2 both have boot fail if pulled low.  So I'm not sure how to find another Pin to use.

![20210604_135002](https://user-images.githubusercontent.com/77108158/120798650-f50ed900-c53d-11eb-9442-6960d47448fc.jpg)
![20210604_135049](https://user-images.githubusercontent.com/77108158/120798665-f9d38d00-c53d-11eb-81c3-376bf0461613.jpg)

**Reply by sfromis** [reply]
Well, I'd not have started with such an unsuitable piece of hardware....

Still, on a small test board, I configured gpio1/3 as suggested by you, did not connect DS18B20, and switched by connecting gpio3 to gnd. No problems. Of course, do not pull low when powering up.

**Reply by Jason2866** [reply]
As you correctly wrote pulling GPIOs 1 & 2 low at boot time will prevent starting the ESP82XX.
This is by design. You can reuse the LED GPIO pin. Desolder LED and use a custom template.
Described here https://templates.blakadder.com/sonoff_basic.html
GPIO 4 & 5 are reachable direct on the mcu since the are in the corners of the chip
